timdex_dataset_api/dataset.py

```diff
@@ -0,0 +1,133 @@
+"""timdex_dataset_api/dataset.py"""
+
+import time
+
+import boto3
+import pyarrow as pa
+import pyarrow.dataset as ds
+from pyarrow import fs
+
+from timdex_dataset_api.config import configure_logger
+from timdex_dataset_api.exceptions import DatasetNotLoadedError
+
+logger = configure_logger(__name__)
+
+TIMDEX_DATASET_SCHEMA = pa.schema(
+    (
+        pa.field("timdex_record_id", pa.string()),
+        pa.field("source_record", pa.binary()),
+        pa.field("transformed_record", pa.binary()),
+        pa.field("source", pa.string()),
+        pa.field("run_date", pa.date32()),
+        pa.field("run_type", pa.string()),
+        pa.field("run_id", pa.string()),
+        pa.field("action", pa.string()),
+    )
+)
+
+
+class TIMDEXDataset:
+
+    def __init__(self, location: str | list[str]):
+        self.location = location
+        self.dataset: ds.Dataset = None  # type: ignore[assignment]
+        self.schema = TIMDEX_DATASET_SCHEMA
+
+    @classmethod
+    def load(cls, location: str) -> "TIMDEXDataset":
+        """Return an instantiated TIMDEXDataset object given a dataset location.
+
+        Argument 'location' may be a local filesystem path or an S3 URI to a parquet
+        dataset.
+        """
+        timdex_dataset = cls(location=location)
+        timdex_dataset.dataset = timdex_dataset.load_dataset()
+        return timdex_dataset
+
+    @staticmethod
+    def get_s3_filesystem() -> fs.FileSystem:
+        """Instantiate a pyarrow S3 Filesystem for dataset loading."""
+        session = boto3.session.Session()
+        credentials = session.get_credentials()
```

Comment:
Curious if `session.get_credentials` retrieves the AWS credentials from the `.env` file (e.g., suggested by [item 3 in this doc](https://boto3.amazonaws.com/v1/documentation/api/latest/guide/credentials.html#configuring-credentials))? 🤔

Reply:
I believe it should.  It's a `boto3` thing, which I think is good at figuring out where possible credentials are.